reflectable/lib/capability.dart

```diff
@@ -518,11 +518,11 @@ class _StringInvocation extends StringInvocation {
 /// doesn't have the capabilities to invoke it.
 class ReflectableNoSuchMethodError extends Error
     implements NoSuchCapabilityError {
-  final Object receiver;
+  final Object? receiver;
```

Comment:
Could we add a line of documentation explaining what it means for this to be null?

Reply:
That was indeed useful: A static method uses `reflectedType` as the receiver, but that would throw when no capability was requested to have that `reflectedType`, so I changed it to test `hasReflectedType` and use null if not. The remaining cases look ok.